### PR TITLE
Project name validation

### PR DESCRIPTION
**Pull request by lcampos**
### What does this PR do?
As part of rewriting the template commands we introduced a validation in the name attribute for the project:create command which caused some regressions. This removes the validation in order to fix the regression.
 
### What issues does this PR fix or reference?
@W-6858055@, #45

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,6 +33,9 @@ testsoutput/
 testpage/
 folder space/
 foo*
+analytics1/
```
Comment (lcampos) [inline .gitignore]: Add missing folders used by tests

File: package.json [hunk 2]
```diff
@@ -27,8 +27,8 @@
     "@types/node": "^10",
     "@types/yeoman-generator": "^3.1.3",
     "chai": "^4",
-    "eslint": "6.0.1",
-    "eslint-config-prettier": "6.0.0",
+    "eslint": "6.6.0",
```
Comment (lcampos) [inline package.json]: Update dev dependency

File: test/commands/project/create.test.ts [hunk 3]
```diff
@@ -153,6 +154,70 @@ describe('Project creation tests:', () => {
           }
         }
       );
+
```
Comment (lcampos) [inline test/commands/project/create.test.ts]: Add a test that verifies users can create a project with a dash as part of the name.

File: test/commands/project/create.test.ts [hunk 4]
```diff
@@ -361,69 +426,5 @@ describe('Project creation tests:', () => {
       .it('should throw invalid template name error', ctx => {
         expect(ctx.stderr).to.contain(messages.getMessage('InvalidTemplate'));
       });
-    test
```
Comment (lcampos) [inline test/commands/project/create.test.ts]: Remove tests specific to project name validations.

File: src/commands/force/project/create.ts [hunk 5]
```diff
@@ -61,7 +61,6 @@ export default class Project extends TemplateCommand {
     })
   };
   public async run(): Promise<AnyJson> {
-    CreateUtil.checkInputs(this.flags.projectname);
```
Comment (lcampos) [inline src/commands/force/project/create.ts]: By removing this validations we now rely on the OS being the one that "checks" if the project name is valid or not, which is what we used to do.

File: .gitignore (hunk 1 above)
Comment (brpowell) [reply .gitignore]: I think you can just add `testsoutput/` to cover anything generated by the tests so we don't have to add individual folders.

Comment (lcampos) [reply .gitignore]: I'll check but I think the ones I added were created outside of `testoutput`

**Pull request merged**